invenio_communities/cache/cache.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Abstract simple identity cache definition."""
+from flask import current_app
+from werkzeug.utils import cached_property
+
+
+class IdentityCache(object):
+    """Abstract cache layer."""
+
+    def __init__(self, app=None):
+        """Initialize the cache."""
+
+    @cached_property
+    def timeout(self):
+        """Return default timeout from config."""
+        return current_app.config["COMMUNITIES_IDENTITIES_CACHE_TIME"]
+
+    def get(self, key):
+        """Return the key value.
+
+        :param key: the object's key
+        """
```

Comment:
maybe a good idea to add `pass` or `raise NotImplementedError`? or maybe mark them as `abstract`?